### ISSUE TITLE
terrain billboarded trees render behind the water

### ISSUE THREAD

**Issue by joaquingrech** [Moonlit-Games/Ultimate-Water-System#46]
Hi

How do we change the render queue so that the water does not render on top of the terrain billboards?
Billboard trees use a transparent shader and it renders behind the water right now.  This is a big problem. I've tried changing the render queue on the shader template but they will always render on top of the trees. I attach a screenshot.

![billboard](https://user-images.githubusercontent.com/1922721/35653349-1d8e768c-06e8-11e8-83e9-8d205f346d58.jpg)

Any help appreciated. This is the only thing stopping me from release right now.

**Comment by MCPGNZ** [comment]
Hi, 
What Unity version are you using? 
I've did some test on Unity 2017.1.1p3.
Using standard Unity-Environment-Asset trees with the _Nature/SpeedTree Billboard_ shader works without any issues on all the render modes:

![image](https://user-images.githubusercontent.com/8217782/36030864-19212582-0da9-11e8-8cef-50bb09193bc0.png)

If you're using some custom shader for rendering, is it from the unity or something else altogether?

**Comment by joaquingrech** [comment]
latest unity 2017.3.1 but it happens with every single unity version. Don't use Speedtrees. Use standard unity trees with the tree creator. Speedtrees are not part of unity and will be removed in near future release due to slow to snail speed performance. (funny they are called speed trees).

**Comment by MCPGNZ** [comment]
If you could send me a sample scene it would be great.
I've used Trees from Unity Tree Editor, and messed with different shaders (UI/Transparent, Standard with Transparency, Alpha Billboards [from this thread](https://forum.unity.com/threads/help-with-billboard-alpha-shader.183371/#post-1254108)) and all of them appear to work correctly with all of the render modes, i think I'm missing something and i do not know what is it, so simple scene would help me a lot.

![image](https://user-images.githubusercontent.com/8217782/36040003-75a63dc2-0dc4-11e8-9e8b-15808297f298.png)

![image](https://user-images.githubusercontent.com/8217782/36040042-8b970f94-0dc4-11e8-8cbb-1e388fddd906.png)

**Comment by MCPGNZ** [comment]
Thanks for the files, 
I've managed to replicate the problem with _Default Queue_ mode; 
you can try to go to the _Water Camera_ script and change the _Render Mode_ to _Image Effect Forward_ or _Image Effect Deferred_.

I'll look into the problem with default queue (but you can tell me if the above mentioned solution will suit your project).

**Comment by joaquingrech** [comment]
nope, in my case using any other render mode gives other troubles with several shaders i have. Only one that works is Default Queue.

**Comment by joaquingrech** [comment]
why is the bug on "waiting" ?
